package.json

```diff
@@ -95,7 +101,7 @@
     "require-in-the-middle": "^7.1.1",
     "rimraf": "^5.0.1",
     "semantic-release": "^21.0.7",
-    "tsup": "^6.7.0",
+    "tsup": "^8.4.0",
```

Comment:
main change 2️⃣ 

Newer version of tsup now correctly creates `d.cts` and `d.ts` files when the config defines that we should emit `['cjs', 'esm']`